azurerm/resource_arm_hdinsight_test.go

```diff
@@ -0,0 +1,171 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMHDInsightCluster_basic(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMHDInsightCluster_basic(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMHDInsightClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHDInsightClusterExists("azurerm_hdinsight_cluster.test"),
+				),
+			},
+		},
+	})
+}
+
+func testAccAzureRMHDInsightCluster_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+		resource "azurerm_resource_group" "test" {
+			name     = "hdinsight-cluster-%d"
+			location = "%s"
+		  
+			tags {
+			  Source = "Azure Quickstarts for Terraform"
+			}
+		  }
+		  
+		  resource "azurerm_storage_account" "test" {
+			  name	= "hdi%d"
```

Comment:
there's a limit on the number of chars available for a storage account name; could we make this a random string of 4-6 chars + with a `acctestsa` prefix